tensorflow/core/util/zen_util.h

```diff
@@ -0,0 +1,198 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_CORE_UTIL_ZEN_UTIL_H_
+#define TENSORFLOW_CORE_UTIL_ZEN_UTIL_H_
+
+#ifdef AMD_ZENDNN
+
+#include <sys/types.h>
+
+#include <array>
+#include <cassert>
+#include <chrono>
+#include <cstdio>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace tensorflow {
+
+/// Read an integer from the environment variable
+/// Return default_value if the environment variable is not defined, otherwise
+/// return actual value.
+inline int zendnn_getenv_int(const char *name, int default_value = 0) {
+  char *val = std::getenv(name);
```

Comment:
Syscalls are expensive. We want to read the environment variable only once and cache the results for subsequent calls. Please follow the approach in `IsZenDnnDisabled()`.